ejb3/src/main/java/org/jboss/as/ejb3/remote/EJBRemoteTransactionPropagatingInterceptor.java

```diff
@@ -114,7 +114,7 @@ private void createOrResumeXidTransaction(final XidTransactionID xidTransactionI
         } else {
             // begin a new tx and add it to the tx repository
             // TODO: Fix the tx timeout to accept a value from the client (WFLY-2789)
-            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.MAX_VALUE);
+            final Transaction newSubOrdinateTx = this.ejbRemoteTransactionsRepository.importTransaction(xidTransactionID, Integer.parseInt(System.getProperty("remote-tx-timeout", "31536000")));
```

Comment:
how does this propagates timeout? who sets this system property?
Propagate would imply that it is taken from somewhere else so is configured in common place.
It should probably be passed by as value in InterceptorFactoryContext which should be configurable in some place.
Maybe even by introducing new attribute in mgmt model.